src/index.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  Beautifier,
+  Language,
+  BeautifierBeautifyData,
+  DependencyType,
+  ExecutableDependency,
+} from "unibeautify";
+import * as readPkgUp from "read-pkg-up";
+import * as tmp from "tmp";
+import * as fs from "fs";
+import * as path from "path";
+import * as os from "os";
+
+const { pkg } = readPkgUp.sync({ cwd: __dirname });
+
+export const beautifier: Beautifier = {
+  name: "clang-format",
+  package: pkg,
+  badges: [
+    {
+      description: "Build Status",
+      url:
+        "https://travis-ci.com/Unibeautify/beautifier-clang-format.svg?branch=master",
+      href: "https://travis-ci.com/Unibeautify/beautifier-clang-format",
+    },
+    {
+      description: "Test Coverage",
+      url:
+        "https://api.codeclimate.com/v1/badges/8b4e5b19de96f4744ffa/test_coverage",
+      href:
+        "https://codeclimate.com/github/Unibeautify/beautifier-clang-format/test_covera" +
+        "ge",
+    },
+    {
+      description: "Maintainability",
+      url:
+        "https://api.codeclimate.com/v1/badges/8b4e5b19de96f4744ffa/maintainability",
+      href:
+        "https://codeclimate.com/github/Unibeautify/beautifier-clang-format/maintainabi" +
+        "lity",
+    },
+    {
+      description: "Greenkeeper",
+      url:
+        "https://badges.greenkeeper.io/Unibeautify/beautifier-clang-format.svg",
+      href: "https://greenkeeper.io/",
+    },
+  ],
+  options: {
+    "C": true,
+    "C++": true,
+    Java: true,
+    "Objective-C": true
+  },
+  dependencies: [
+    {
+      type: DependencyType.Executable,
+      name: "clang-format",
+      program: "clang-format",
+      parseVersion: [/clang-format version (\d+\.\d+\.\d+)/],
+      homepageUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      installationUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      bugsUrl: "https://clang.llvm.org/docs/ClangFormat.html",
+      badges: [
+        {
+          description: "",
+          url: "",
+          href: "",
+        },
+      ],
+    },
+  ],
+  resolveConfig: ({ filePath, projectPath }) => {
+    const configFiles: string[] = [".clang-format", "_clang-format"];
+    return findFile({
+      finishPath: projectPath,
+      startPath: filePath,
+      fileNames: configFiles,
+    })
+      .then(configFile => ({ filePath: configFile }))
+      .catch(err => {
+        // tslint:disable-next-line no-console
+        console.log(err);
+        return Promise.resolve({});
+      });
+  },
+  beautify({
+    text,
+    dependencies,
+    filePath,
+    beautifierConfig,
+  }: BeautifierBeautifyData) {
+    const clangFormat = dependencies.get<ExecutableDependency>("clang-format");
+    const basePath: string = os.tmpdir();
+    let fileExtension;
+    if (filePath) {
+      fileExtension = path.extname(filePath);
+    }
+    const config =
+      beautifierConfig && beautifierConfig.filePath
+        ? `--style=${beautifierConfig.filePath}`
```

Comment:
Does this work?

Looking at https://clang.llvm.org/docs/ClangFormatStyleOptions.html 

```
  -style=<string>           - Coding style, currently supports:
                                LLVM, Google, Chromium, Mozilla, WebKit.
                              Use -style=file to load style configuration from
                              .clang-format file located in one of the parent
                              directories of the source file (or current
                              directory for stdin).
                              Use -style="{key: value, ...}" to set specific
                              parameters, e.g.:
                                -style="{BasedOnStyle: llvm, IndentWidth: 8}"
```

https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/clang-format.coffee#L82

I keep seeing `--style=file` and how it automatically finds the config file.

Reply:
Also see https://github.com/Glavin001/atom-beautify/blob/master/src/beautifiers/clang-format.coffee#L59

```
    # As indicated in the manual,
    # clang-format (with `--style file`) tries to locate a `.clang-format`
    # config file in directory and parent directories of the input file,
    # and retreat to default style if not found. 
```

Reply:
>                               Use -style="{key: value, ...}" to set specific
                              parameters, e.g.:
                                -style="{BasedOnStyle: llvm, IndentWidth: 8}"


What about passing in the configuration as a string?
If we have a `beautifierConfig.filePath` we `fs.readFile` and pass it in directly.

Ideally, we reduce the need to create and read from temporary files. Pass as much as possible via CLI arguments and stdin and use stdout.

Reply:
Yes we need to pull the contents of the config file and pass directly into the cli as a string.  This PR does not do that.  So for now `-style=file` is in there so clang-format itself looks for it.